app/controllers/questions_controller.rb

```diff
@@ -95,4 +96,13 @@ def questions_property
       QuestionsProperty.new('未解決の質問一覧', '未解決の質問はまだありません。')
     end
   end
+
+  def watching_mentors_only
+    User.mentor.each do |mentor|
+      Watch.create!(
+        user: mentor,
+        watchable: @question
+      )
+    end
+  end
```

Comment:
## 1. メンターの数だけSQLが発行される（N+1）

メンターが10人いると10回INSERT文が発行されて遅くなってしまいます。

[insert_all!](https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-insert_all-21) を使って一回のSQLでレコードを追加するように直す方が良さそうです。

## 2. Watchの登録が途中で終わる可能性がある

メンターが10人いて、5人目のWatch登録でエラーになると残りはWatchにならずにエラー画面が表示されてしまいそうです。
対応方針は @komagata さんに伺った方が良いかも。

1. 現状のまま途中でエラーが出たら、そのままエラーにする
    - Watchの登録に失敗することは稀だと思うので、エラー出てから考えるでも良い気はします
2. エラーを記録しつつ、残りの登録を継続して、最後にエラーを発生させる
    - エラーになったメンターの人を記録するので、コードが少し複雑になる

Reply:
@sinsoku 
アドバイスありがとうございます！
insert_allを使用する形に修正してみました(^^
但し、rubocopにisert_allは使わないようにメッセージが出てしまいましたので、駒形さんに相談して更に対応したい思います。🙇‍♂️
rubocopのメッセージ：`Rails/SkipsModelValidations: Avoid using insert_all because it skips validations.`

@komagata 
遅くなりすみません。「質問を投稿するとメンターのみ"Watch中"にする」機能を修正しました。
メソッド名の変更しました。すみませんが、以下の確認をお願い致します。

- メソッド名：`create_mentors_watch`は妥当でしょうか。
- `insert_all`は使用しない方がよろしいでしょうか。
- sinsokuさんのご指摘のように「メンターが10人いて、5人目のWatch登録でエラーになると残りはWatchにならずにエラー画面が表示されてしまいそう」のような想定をした方がよろしいでしょうか。それとも、別の想定がよろしいでしょうか。

Reply:
@sinsoku 
ご連絡遅くなってすみません。
駒形さんにも伺って、今回はrubocopの指摘を優先することとなりました。
とは言え、アドバイスとても勉強になりました。ありがとうございました。🙇‍♂️

Reply:
@yamadaab 大変申し訳ありません、改めて考えてみたところメンター12人ぐらいいるので速度的にinsert_allを使った方がいいように思いました。

お手数ですがinsert_allにしてrubocopの指摘はコメントを使って無効化していただければありがたいです。

Reply:
@komagata 
時間がかかりすみません。rubocopに手間取りましたが、無事に修正できたと思います。
ご確認よろしくお願い致します。